txpool/pool.go

```diff
@@ -1627,12 +1627,14 @@ func (p *TxPool) flushLocked(tx kv.RwTx) (err error) {
 			continue
 		}
 		v = common.EnsureEnoughSize(v, 20+len(metaTx.Tx.Rlp))
-		for addr, id := range p.senders.senderIDs { // no inverted index - tradeoff flush speed for memory usage
-			if id == metaTx.Tx.SenderID {
-				copy(v[:20], addr)
-				break
-			}
+
+		addr, ok := p.senders.senderID2Addr[metaTx.Tx.SenderID]
+		if !ok {
+			log.Warn("sender address not found by ID", metaTx.Tx.SenderID)
```

Comment:
erigon has many components - plz add prefix `[txpool] flush:` to understand where this log comes from